pkg/lib/oauth/handler/handler_token.go

```diff
@@ -155,6 +155,9 @@ func (h *TokenHandler) doHandle(
 		allowedGrantTypes = []string{"authorization_code"}
 	}
 	allowedGrantTypes = append(allowedGrantTypes, whitelistedGrantTypes...)
+	if client.App2appEnabled {
+		allowedGrantTypes = append(allowedGrantTypes, App2AppRequestGrantType)
```

Comment:
What about adding `App2AppRequestGrantType` to `whitelistedGrantTypes`?

Reply:
But for a app didn't enabled app2app, do we want to allow using app2app grant type?

Reply:
We have done this for anonymous user and biometric login. I think we should follow.